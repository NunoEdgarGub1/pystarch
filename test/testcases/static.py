x01 = 1
x02 = x01 + 2
x03 = False
x04 = x03 and True
x05 = x04 or True


def f(x):
    if True:
        return 2 + x
    else:
        return None

x06 = f(1)
x07 = x06 is not None

if x06 is not None:
    x08 = x06 + 2
else:
    x08 = x06 + 3
