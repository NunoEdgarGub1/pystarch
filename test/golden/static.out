f Function(x: Unknown)
x01 Num 1
x02 Num 3
x03 Bool False
x04 Bool False
x05 Bool True
x06 Maybe(Num)
x07 Bool
x08 Num

testcases/static.py:9 constant-if-condition "If"
testcases/static.py:20 inconsistent-types "Add" (NoneType, Num)
